nessie/src/test/java/org/apache/iceberg/nessie/TestBranchVisibility.java

```diff
@@ -173,6 +174,33 @@ public void testSchemaSnapshot() throws Exception {
     Assertions.assertThat(metadataOn2).isNotEqualTo(metadataOnTest).isNotEqualTo(metadataOnTest2);
   }
 
+  @Test
+  public void testMetadataLocationWithRefresh() throws Exception {
+    String branch1 = "test";
+    String branch2 = "branch-2";
+
+    // commit on tableIdentifier1 on branch1
+    NessieCatalog catalog = initCatalog(branch1);
+    String metadataCommit1 = addRow(catalog, tableIdentifier1, "initial-data",
+        ImmutableMap.of("id0", 4L));
+    catalog.refresh();
+
+    createBranch(branch2, catalog.currentHash(), branch1);
+    // commit on tableIdentifier1 on branch2
+    NessieCatalog catalogBranch1 = initCatalog(branch2);
```

Comment:
maybe just change to `catalog = initCatalog(branch2)`

Reply:
I followed the pattern from base testcases in this file. 
Changed this on newly added testcase, base testcases can be done in a follow up PR as it will confuse the reviewers.